### ISSUE TITLE
Conditional byref assignment is not caught

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#123]
```php
function foo() : void {
  	$b = null;
  	$c = rand(0, 1) ? bar($b) : null;
  	if (is_int($b)) { }
}
function bar(?int &$a) : void {
	$a = 5;
}
```

Emits a `TypeDoesNotContainType` error.
